base/strings/util.jl

```diff
@@ -358,7 +358,10 @@ _replace(io, repl, str, r, pattern) = print(io, repl)
 _replace(io, repl::Function, str, r, pattern) =
     print(io, repl(SubString(str, first(r), last(r))))
 
-function replace(str::String, pattern, repl, limit::Integer)
+function replace_new(str::String, pattern, repl, count::Int)
```

Comment:
The problem is that once the deprecation is removed, the "API level" will be this function, and only `Int` will be accepted... So if you want this you need to add another layer which will remain. But really I'm not sure it's worth it, e.g. `search` doesn't use that trick.